node_package/src/loadingSpinner/components/MediaLoadingSpinner.jsx

```diff
@@ -0,0 +1,56 @@
+import {combineSelectors} from 'utils';
+
+import React from 'react';
+import {connect} from 'react-redux';
+import classNames from 'classnames';
+
+import registerWidgetType from 'registerWidgetType';
+import {entryAttribute} from 'entry/selectors';
+import {editingWidget, widgetAttribute} from 'widgets/selectors';
+import {file} from 'files/selectors';
+import {pageBackgroundImageUrl, firstPageAttribures} from 'pages/selectors';
+import {TitleLoadingSpinner, preventScrollBouncing, inlineStyle, backgroundImageInlineStyles} from './TitleLoadingSpinner';
+
+class MediaLoadingSpinner extends TitleLoadingSpinner {
+  render() {
+    const {hidden, animating} = this.state;
+    var logoElement = <div className="title_loading_spinner-logo" />
+    if (this.props.removeLogo) {
+      logoElement = ''
+    }
+    if (this.props.editing || !hidden) {
+      return (
+        <div className={classNames('title_loading_spinner', {'title_loading_spinner-fade': animating})}
+             onAnimationEnd={(event) => this.hideOrLoop(event)}
+             onTouchMove={preventScrollBouncing}
+             style={inlineStyle()}>
+          {logoElement}
+          <div className="title_loading_spinner-image"
+               style={backgroundImageInlineStyles(this.props)} />
+        </div>
+      );
+    }
+    else {
+      return <noscript />;
+    }
+  }
+}
```

Comment:
Inheritance is rather unidiomatic in React. The overridden `render` method requires a lot of formerly private functions of `TitleLoadingSpinner` to be exported which creates further dependencies and weakens encapsulation. 

Instead, I'd suggest to go with composition. Comparing two render methods I only see a few differences:

* `MediaLoadingSpinner` allows removing the logo. Any reason not to allow this for `TitleLoadingSpinner`?
* `TitleLoadingSpinner` supports invert option. If the logo is not removed, wouldn't this also be required for `MediaLoadingSpinner`?
* `TitleLoadingSpinner` contains additional elements for the title elements.

I'd therefore suggest to make `MediaLoadingSpinner` render `props.children` right after the background image div. Then `TitleLoadingSpinner` can use composition to render a `MediaLoadingSpinner` component passing the additional title elements as children.

`TitleLoadingSpinner` can still be connected separately to ask for additional widget attributes like `title` and `subtitle`, but all the now duplicated "map state to props" entries (editing, firstPageBackgroundImage. backgroundImage, ..) only need to live in `TitleLoadingSpinner`.

Since now media loading spinner is the more basic concept, we should consider renaming the CSS classes that have a `title_loading_spinner-` prefix to `media_loading_spinner-*`.

Reply:
Refactored MediaLoadingSpinner and TitleLoadingSpinner.
MediaLoadingSpinner is now composed by TitleLoadingSpinner using its connect component